File: news/src/main/java/com/leory/badminton/news/mvp/ui/fragment/NewMainFragment.java
package com.leory.badminton.news.mvp.ui.fragment;

import android.content.Context;
import android.graphics.Color;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v4.view.ViewPager;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.alibaba.android.arouter.facade.annotation.Route;
import com.leory.badminton.news.R;
import com.leory.badminton.news.R2;
import com.leory.badminton.news.mvp.ui.adapter.IndicatorPagerAdapter;
import com.leory.commonlib.base.BaseFragment;
import com.leory.commonlib.utils.StatusBarUtils;
import com.leory.interactions.RouterHub;

import net.lucode.hackware.magicindicator.MagicIndicator;
import net.lucode.hackware.magicindicator.ViewPagerHelper;
import net.lucode.hackware.magicindicator.buildins.commonnavigator.CommonNavigator;
import net.lucode.hackware.magicindicator.buildins.commonnavigator.abs.CommonNavigatorAdapter;
import net.lucode.hackware.magicindicator.buildins.commonnavigator.abs.IPagerIndicator;
import net.lucode.hackware.magicindicator.buildins.commonnavigator.abs.IPagerTitleView;
import net.lucode.hackware.magicindicator.buildins.commonnavigator.titles.ClipPagerTitleView;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

import butterknife.BindView;

/**
 * Describe : 资讯首页fragment
 * Author : zhuly
 * Date : 2019-05-22
 */
@Route(path = RouterHub.NEWS_NEWMAINFRAGMENT)
public class NewMainFragment extends BaseFragment {

    @BindView(R2.id.indicator)
    MagicIndicator indicator;
    @BindView(R2.id.view_pager)
    ViewPager viewPager;
    private static final String[] CHANNELS = new String[]{"直播", "赛事", "排名", "球星"};
    private List<Fragment> fragmentList = new ArrayList<>();
    private List<String> dataList = Arrays.asList(CHANNELS);
    private IndicatorPagerAdapter indicatorPagerAdapter;


    @Override
    public View initView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_news_main, container, false);
    }

    @Override
    public void initData(@Nullable Bundle savedInstanceState) {
        StatusBarUtils.setDarkStatusBar(getActivity(), true, R.color.colorPrimary);//进来没有执行onPageSelected回调，这里要设置一下
        fragmentList.add(new LiveFragment());
        fragmentList.add(new MatchFragment());
        fragmentList.add(new RankingFragment());
        fragmentList.add(new RankingFragment());
        indicatorPagerAdapter = new IndicatorPagerAdapter(getChildFragmentManager(), fragmentList);
        viewPager.setAdapter(indicatorPagerAdapter);
        viewPager.setOffscreenPageLimit(4);
        initIndicator();
    }


    private void initIndicator() {
        indicator.setBackgroundResource(R.color.colorPrimary);
        CommonNavigator commonNavigator = new CommonNavigator(getActivity());
        commonNavigator.setSkimOver(true);
        commonNavigator.setAdapter(new CommonNavigatorAdapter() {

            @Override
            public int getCount() {
                return dataList == null ? 0 : dataList.size();
            }

            @Override
            public IPagerTitleView getTitleView(Context context, final int index) {
                ClipPagerTitleView clipPagerTitleView = new ClipPagerTitleView(context);
                clipPagerTitleView.setText(dataList.get(index));
                clipPagerTitleView.setTextColor(Color.parseColor("#f2c4c4"));
                clipPagerTitleView.setClipColor(Color.WHITE);
                clipPagerTitleView.setOnClickListener(new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        viewPager.setCurrentItem(index);
                    }
                });
                return clipPagerTitleView;
            }

            @Override
            public IPagerIndicator getIndicator(Context context) {
                return null;
            }
        });
        indicator.setNavigator(commonNavigator);
        ViewPagerHelper.bind(indicator, viewPager);
    }

}

File: news/src/main/java/com/leory/badminton/news/mvp/ui/fragment/NewMainFragment.kt
package com.leory.badminton.news.mvp.ui.fragment

import android.content.Context
import android.graphics.Color
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.alibaba.android.arouter.facade.annotation.Route
import com.leory.badminton.news.R
import com.leory.badminton.news.mvp.ui.adapter.IndicatorPagerAdapter
import com.leory.commonlib.base.BaseFragment
import com.leory.commonlib.mvp.IPresenter
import com.leory.commonlib.utils.StatusBarUtils
import com.leory.interactions.RouterHub
import kotlinx.android.synthetic.main.fragment_news_main.*
import net.lucode.hackware.magicindicator.ViewPagerHelper
import net.lucode.hackware.magicindicator.buildins.commonnavigator.CommonNavigator
import net.lucode.hackware.magicindicator.buildins.commonnavigator.abs.CommonNavigatorAdapter
import net.lucode.hackware.magicindicator.buildins.commonnavigator.abs.IPagerIndicator
import net.lucode.hackware.magicindicator.buildins.commonnavigator.abs.IPagerTitleView
import net.lucode.hackware.magicindicator.buildins.commonnavigator.titles.ClipPagerTitleView
import java.util.*

/**
 * Describe : 资讯首页fragment
 * Author : zhuly
 * Date : 2019-05-22
 */
@Route(path = RouterHub.NEWS_NEWMAINFRAGMENT)
class NewMainFragment : BaseFragment<IPresenter>() {
    companion object {
        private val CHANNELS = arrayOf("直播", "赛事", "排名", "球星")
    }

    private val fragmentList = ArrayList<Fragment>()
    private val dataList = listOf(*CHANNELS)
    private var indicatorPagerAdapter: IndicatorPagerAdapter? = null


    override fun initView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        return inflater.inflate(R.layout.fragment_news_main, container, false)
    }

    override fun initData(savedInstanceState: Bundle?) {
        StatusBarUtils.setDarkStatusBar(activity, true, R.color.colorPrimary)//进来没有执行onPageSelected回调，这里要设置一下
        fragmentList.add(LiveFragment())
        fragmentList.add(RankingFragment())
        fragmentList.add(RankingFragment())
        fragmentList.add(RankingFragment())
        indicatorPagerAdapter = IndicatorPagerAdapter(childFragmentManager, fragmentList)
        view_pager.adapter = indicatorPagerAdapter
        view_pager.offscreenPageLimit = 4
        initIndicator()
    }


    private fun initIndicator() {
        indicator!!.setBackgroundResource(R.color.colorPrimary)
        val commonNavigator = CommonNavigator(activity)
        commonNavigator.isSkimOver = true
        commonNavigator.adapter = object : CommonNavigatorAdapter() {

            override fun getCount(): Int {
                return dataList?.size ?: 0
            }

            override fun getTitleView(context: Context, index: Int): IPagerTitleView {
                val clipPagerTitleView = ClipPagerTitleView(context)
                clipPagerTitleView.text = dataList[index]
                clipPagerTitleView.textColor = Color.parseColor("#f2c4c4")
                clipPagerTitleView.clipColor = Color.WHITE
                clipPagerTitleView.setOnClickListener { view_pager.currentItem = index }
                return clipPagerTitleView
            }

            override fun getIndicator(context: Context): IPagerIndicator? {
                return null
            }
        }
        indicator!!.navigator = commonNavigator
        ViewPagerHelper.bind(indicator, view_pager)
    }


}
